tests/migrations/test_commands.py

```diff
@@ -2082,6 +2082,29 @@ def test_squashed_name_without_start_migration_name(self):
             squashed_migration_file = os.path.join(migration_dir, '0001_%s.py' % squashed_name)
             self.assertTrue(os.path.exists(squashed_migration_file))
 
+    def test_squashed_name_collision(self):
+        """Using --squashed-name shouldn’t overwrite existing migrations"""
+        initial_migration_name = 'initial'
+        with self.temporary_migration_module(module='migrations.test_migrations') as migration_dir:
+            initial_migration_file = os.path.join(
+                migration_dir, '0001_%s.py' % initial_migration_name
+            )
+            with open(initial_migration_file, 'rt') as f:
+                initial_migration_contents = f.read()
+
+            with self.assertRaisesMessage(
+                CommandError, 'not safe to overwrite'
+            ):
+                call_command(
+                    'squashmigrations', 'migrations', '0001', '0002',
+                    squashed_name=initial_migration_name, interactive=False,
+                    verbosity=0,
+                )
+
+            with open(initial_migration_file, 'rt') as f:
+                updated_migration_contents = f.read()
+            self.assertEqual(initial_migration_contents, updated_migration_contents)
```

Comment:
IMO there is no need to check a file content:
```suggestion
            msg = '...'
            with self.assertRaisesMessage(CommandError, msg):
                call_command(
                    'squashmigrations', 'migrations', '0001', '0002',
                    squashed_name='initial', interactive=False, verbosity=0,
                )
```

Reply:
Ok, I can do that. I included the file comparison because the previous behaviour was to overwrite the existing migration file with one that would usually trigger a `CircularDependencyError`.